src/cluster_legacy.c

```diff
@@ -3224,12 +3233,12 @@ int clusterProcessPacket(clusterLink *link) {
             setClusterNodeToInboundClusterLink(node, link);
             clusterAddNode(node);
             clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG);
-        }
 
-        /* If this is a MEET packet from an unknown node, we still process
-         * the gossip section here since we have to trust the sender because
-         * of the message type. */
-        if (!sender && type == CLUSTERMSG_TYPE_MEET) clusterProcessGossipSection(hdr, link);
+            /* If this is a MEET packet from an unknown node, we still process
+             * the gossip section here since we have to trust the sender because
+             * of the message type. */
+            clusterProcessGossipSection(hdr, link);
+        }
```

Comment:
We don't need this change.

Reply:
True, but this double if with the same condition was driving me crazy.

Reply:
I don't mind this. But in general we avoid changes to the lines of code not related to the PR.